Osiris/Hacks/Animations.cpp

```diff
@@ -0,0 +1,154 @@
+#include "Animations.h"
+
+#include "../Memory.h"
+#include "../Interfaces.h"
+
+#include "../SDK/Engine.h"
+#include "../SDK/GlobalVars.h"
+#include "../SDK/LocalPlayer.h"
+#include "../SDK/ModelInfo.h"
+
+std::array<PlayerHook, 65> playerHooks;
+std::array<AnimData, 65> animData;
+
+static Vector* __FASTCALL getEyeAngles(Entity* entity, uint32_t edx) noexcept
+{
+	static auto oGetEyeAngles = playerHooks[entity->index()].vmt.getOriginal<Vector*, 170>();
+
+	if (entity->index() != localPlayer->index() || RETURN_ADDRESS() != memory->setEyePositionAndVectors)
+		return oGetEyeAngles(entity);
+
+	return &animData[entity->index()].sentViewAngles;
+}
+
+static void __FASTCALL buildTransformations(Entity* entity, uint32_t edx, StudioHdr* hdr, Vector* pos, void* q, matrix3x4* cameraTransform, int boneMask, void* boneComputed) noexcept
+{
+	static auto oBuildTransformations = playerHooks[entity->index()].vmt.getOriginal<void, 190, StudioHdr*, Vector*, void*, matrix3x4*, int, void*>(hdr, pos, q, cameraTransform, boneMask, boneComputed);
+
+	std::array<int, 256> backupFlags;
+	for (int i = 0; i < 256; i++) {
+		if (auto bone = hdr->getBone(i); bone) {
+			backupFlags[i] = bone->flags;
+			bone->flags &= ~0x04;
+		}
+	}
+
+	oBuildTransformations(entity, hdr, pos, q, cameraTransform, boneMask, boneComputed);
+
+	for (int i = 0; i < 256; i++) {
+		if (auto bone = hdr->getBone(i); bone) {
+			bone->flags = backupFlags[i];
+		}
+	}
+}
+
+static void __FASTCALL doExtraBoneProcessing(Entity* entity, uint32_t edx, StudioHdr* hdr, Vector* pos, void* q, matrix3x4* matrix, void* boneComputed, void* context) noexcept
+{
+	static auto oDoExtraBoneProcessing = playerHooks[entity->index()].vmt.getOriginal<void, 198, StudioHdr*, Vector*, void*, matrix3x4*, void*, void*>(hdr, pos, q, matrix, boneComputed, context);
+	const auto backupOnGround = entity->getAnimstate()->onGround;
+
+	entity->getAnimstate()->onGround = false;
+	oDoExtraBoneProcessing(entity, hdr, pos, q, matrix, boneComputed, context);
+	entity->getAnimstate()->onGround = backupOnGround;
```

Comment:
No need to set onGround to false, just return, doExtraBoneProcessing isnt called on the server